File: project/CCNet.CSharpDemos.Plugin/Tasks/HelloWorldTaskResult.cs
namespace CCNet.CSharpDemos.Plugin.Tasks
{
    using ThoughtWorks.CruiseControl.Core;

    public class HelloWorldTaskResult
        : ITaskResult
    {
        private readonly string personName;
        private readonly IIntegrationResult result;

        public HelloWorldTaskResult(string personName, IIntegrationResult result)
        {
            this.personName = personName;
            this.result = result;
        }

        public string Data
        {
            get
            {
                return "Hello " + this.personName +
                    " from " + this.result.ProjectName +
                    "(build started " + this.result.StartTime.ToString() + ")";
            }
        }

        public bool CheckIfSuccess()
        {
            return true;
        }
    }
}

File: project/CCNet.VBDemos.Plugin/Tasks/HelloWorldTaskResult.vb
Imports ThoughtWorks.CruiseControl.Core

Public Class HelloWorldTaskResult
    Implements ITaskResult

    Private ReadOnly myPersonName As String
    Private ReadOnly myResult As IIntegrationResult

    Public Sub New(ByVal personName As String, ByVal result As IIntegrationResult)
        myPersonName = personName
        myResult = result
    End Sub

    Public Function CheckIfSuccess() As Boolean _
    Implements ITaskResult.CheckIfSuccess
        Return True
    End Function

    Public ReadOnly Property Data() As String _
    Implements ITaskResult.Data
        Get
            Return "Hello " & myPersonName & _
                    " from " & myResult.ProjectName & _
                    "(build started " & myResult.StartTime.ToString() & ")"
        End Get
    End Property
End Class
